frontend/lib/resolution/split-init.cpp

```diff
@@ -276,22 +278,21 @@ void FindSplitInits::handleConditional(const Conditional* cond, RV& rv) {
   // split init is OK if:
   //   both then & else blocks initialize it before mentioning it
   //   one initializes it and the other returns
+  //   one initializes and its statically known to be the only path
   for (const auto& id : locInitedVars) {
-    bool thenInits = false;
-    if (thenFrame) {
-      thenInits = thenFrame->initedVars.count(id) > 0;
-    }
-    bool elseInits = false;
-    if (elseFrame) {
-      elseInits = elseFrame->initedVars.count(id) > 0;
-    }
+    
+    bool thenInits = thenFrame && thenFrame->initedVars.count(id) > 0;
+    bool elseInits = elseFrame && elseFrame->initedVars.count(id) > 0;
```

Comment:
Something tells me that the same trick from the copy elision section (getting a list of frames and iterating over them; using a map to count initialization and mentions) can help unify this code and the code for select statements. It would also, I think, avoid the `O(n^2)` iteration for finding if a mention occurred somewhere else. I think it's worthwhile to give that a go, because both the if-else and select code are relatively long and difficult to get through. If they are unified, at least there's only one entry point for complex split-init logic.